galaxy/tools/deps/mulled/get_tests.py

```diff
@@ -0,0 +1,263 @@
+	['recipes/samtools/0.1.12/meta.yaml', 'recipes/samtools/0.1.16/meta.yaml', 'recipes/samtools/0.1.17/meta.yaml', 'recipes/samtools/0.1.14/meta.yaml', 'recipes/samtools/0.1.13/meta.yaml', 'recipes/samtools/0.1.15/meta.yaml', 'recipes/samtools/0.1.19/meta.yaml', 'recipes/samtools/1.0/meta.yaml', 'recipes/samtools/0.1.18/meta.yaml', 'recipes/samtools/1.1/meta.yaml']
+
+	"""
+	if RECIPES_REPO_PATH:
+		return [n.replace('%s/' % RECIPES_REPO_PATH, '') for n in glob('%s/%s/*/%s' % (RECIPES_REPO_PATH, filepath, filename))]
+	# else use the GitHub API:
+	versions = []
+	r = json.loads(requests.get('https://api.github.com/repos/bioconda/bioconda-recipes/contents/%s' % filepath).content)
+	for subfile in r:
+		if subfile['type'] == 'dir':
+			if requests.get('https://raw.githubusercontent.com/bioconda/bioconda-recipes/master/%s/%s' % (subfile['path'], filename)).status_code == 200:
+				versions.append('%s/%s' % (subfile['path'], filename))
+	return versions
+
+
+def get_test(container):
+    name = container.replace('--', ':').split(':')
+
+    # first try meta.yaml in correct version folder
+
+    try:
+        t = get_commands_from_yaml(open_recipe_file('recipes/%s/%s/meta.yaml' % (name[0], name[1])))
+    except IOError: 
+        logging.info('/home/ubuntu/GitRepos/bioconda-recipes/recipes/%s/%s/meta.yaml could not be opened.' % (name[0], name[1]))
+        t = None
+    if t:
+        t['container'] == container
+        return t
+
+    # try run_test in base folder
+
+    try:
+        t = get_runtest(open_recipe_file('recipes/%s/%s/run_test.sh' % (name[0], name[1])))
+    except IOError: 
+        logging.info('/home/ubuntu/GitRepos/bioconda-recipes/recipes/%s/%s/run_test.sh could not be opened.' % (name[0], name[1]))
+        t = None
+    if t:
+        t['container'] == container
+        return t
+
+    # now try meta.yaml in base folder
+    try:
+        t = get_commands_from_yaml(open_recipe_file('recipes/%s/meta.yaml' % name[0]))
+    except IOError: 
+        logging.info('/home/ubuntu/GitRepos/bioconda-recipes/recipes/%s/meta.yaml could not be opened.' % name[0])
+        t = None
+    if t:
+        t['container'] == container
+        return t
+
+    # try run_test in base folder
+
+    try:
+        t = get_runtest(open_recipe_file('recipes/%s/run_test.sh' % name[0]))
+    except IOError: 
+        logging.info('/home/ubuntu/GitRepos/bioconda-recipes/recipes/%s/run_test.sh could not be opened.' % name[0])
+        t = None
+    if t:
+        t['container'] == container
+        return t
+
+    # try from anaconda
+
+    t = get_test_from_anaconda(get_anaconda_url(container))
+    if t:
+        t['container'] == container
+        return t
+    logging.info('Nothing on anaconda.')
+    # now try in incorrect version folders
+
+    #g = glob('/home/ubuntu/GitRepos/bioconda-recipes/recipes/%s/*/meta.yaml' % container)
+    g = get_alternative_versions('recipes/%s' % container, 'meta.yaml')
+    for n in g:
+        try:
```

Comment:
This try/except blog with the check and the assignment of the container, can probably live in its own function, right?
You could pass the function and the container name into the function and return the filled dict.